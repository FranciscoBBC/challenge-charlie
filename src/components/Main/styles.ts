import styled from 'styled-components';

export const Container = styled.main<{ bgImage: string }>`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100vh;
    background: center / cover no-repeat
            linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)),
        url(${({ bgImage }) => bgImage || ''});
`;
