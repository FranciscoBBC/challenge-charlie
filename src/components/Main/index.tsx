import React from 'react';
import * as S from './styles';
// imagem mockada para testar o componente
import sunset from '../../assets/img/sunset.jpg';

type MainProps = {
    children: React.ReactNode;
};

const Main = ({ children }: MainProps) => {
    return <S.Container bgImage={sunset}>{children}</S.Container>;
};

export default Main;
