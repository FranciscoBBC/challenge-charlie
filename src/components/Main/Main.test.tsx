import React from 'react';
import { render, screen } from '@testing-library/react';
import Main from '.';

describe('<Main />', () => {
    it('should be render children', () => {
        const { container } = render(
            <Main>
                <p>testing</p>
            </Main>
        );
        expect(screen.getByText(/testing/i)).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should be render background', () => {
        const { container } = render(<Main>teste</Main>);

        expect(container.firstChild).toHaveStyle({
            background: 'linear-gradient(#6698CB, #69B4C9)'
        });
    });
});
