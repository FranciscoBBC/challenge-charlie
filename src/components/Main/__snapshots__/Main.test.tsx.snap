// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should be render children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100vh;
  background: linear-gradient(#6698CB,#69B4C9);
}

<main
  class="c0"
  data-testid="bg-image"
>
  <p>
    testing
  </p>
</main>
`;
